src/pipx/util.py

```diff
@@ -174,7 +174,7 @@ def run_subprocess(
     # See https://github.com/pypa/pipx/issues/1164
     # Conversely, if the binary is a symlink, then we should NOT use the real path, as Python expects to receive the
     # symlink in argv[0] so that it can locate the venv.
-    if not os.path.islink(cmd_str_list[0]):
+    if not os.path.islink(cmd_str_list[0]) and WINDOWS:
```

Comment:
The file uses a mix of `pathlib.Path` and `os.path`. pathlib supports a islink equivalent (`Path.is_symlink`) and a `realpath` (`Path.resolve`). I think it might make sense to use those rather than os.path for the 2 lines, but I can't really give a compelling reason, so it's ok to completely ignore this too.

Also, before I saw this PR, I was thinking about dealing with it by checking if `cmd_str_list[0]` is absolute or not, but your solution `and WINDOWS` also makes sense. I wanted to suggest the alternative because I don't know yet if any is better, but other reviewers may have ideas on the matter. I do think the `and WINDOWS` approach is at least very pragmatic and unlikely to break further so 👍 

Finally, I think the line would be more readable as:
```suggestion
    if WINDOWS and not os.path.islink(cmd_str_list[0]):
```
because then when you read the `is_link` condition, you already know you're in the windows and...

I wonder if `islink` can actually return True under windows, so is it possible that this actually is just:
```suggestion
    if WINDOWS:
```
?

I'll try to inspect the cpython source. I don't know enough Windows to tell :D

Reply:
CPython under Windows [has an implementation](https://github.com/python/cpython/blob/86ebd5c3fa9ac0fba3b651f1d4abfca79614af5f/Modules/posixmodule.c#L5321) for islink

Though it's unclear if it makes sense to check for symlink if we know we're under windows, as I'm unsure venvs use symlinks under windows.

Reply:
The original fix author, @dechamps might have an opinion?

Reply:
> I'm unsure whether venvs use symlinks under windows

They don't unless you explicitly tell `venv` to use symlinks, see [this doc](https://docs.python.org/3/library/venv.html#creating-virtual-environments) and especially the note "While symlinks are supported on Windows, they are not recommended". As long as it's pipx that's creating the venv, we can assume that it's not using symlinks on Windows as pipx does not pass `--symlinks` to `venv`.

I think this change makes sense, and I agree it can probably be simplified to just `if WINDOWS:`. This is probably what I should have written from the start - looks like I was a bit too obsessed with making this somewhat platform-agnostic and ended up breaking things 😢

Reply:
Don't beat yourself up too hard, that stuff is too complex for any sane person to navigate with ease. You fixed your case, and wrote some test, nothing more may really be expected from anyone. Sometimes, luck factors in, too. And thanks for chiming in on such a short notice <3

Reply:
Thank you for the review @ewjoachim and thanks for chiming in @dechamps :) 

Will simplify the if statement and clean up the commend above of it after work :+1:

Reply:
(Just to be extremely explicit: given the conditions, I believe it's not worth switching the single `realpath` to `pathlib`, I withdraw my suggestion regarding that part)